openstf/model/confidence_interval_applicator.py

```diff
@@ -7,19 +7,29 @@
 import numpy as np
 import pandas as pd
 from scipy import stats
+from sklearn.base import RegressorMixin
+
+from openstf.enums import MLModelType
 
 MINIMAL_RESOLUTION: int = 15  # Minimal time resolution in minutes
 
 
 class ConfidenceIntervalApplicator:
-    def __init__(self, model):
-        self.confidence_interval = model.confidence_interval
+    def __init__(self, model: RegressorMixin, forecast_input_data: pd.DataFrame):
+        self.model = model
+        self.forecast_input_data = forecast_input_data
 
-    def add_confidence_interval(
-        self, forecast: pd.DataFrame, quantiles: list
-    ) -> pd.DataFrame:
+    def add_confidence_interval(self, forecast: pd.DataFrame, pj: dict) -> pd.DataFrame:
         temp_forecast = self._add_standard_deviation_to_forecast(forecast)
-        return self._add_quantiles_to_forecast(temp_forecast, quantiles)
+
+        if pj["model"] == MLModelType.XGB_QUANTILE:
```

Comment:
Zou het logisch zijn als het 'type-confidence-interval' een attribuut van het `model` zou zijn? Zodat je bij de definitie van een model opgeeft dat het bijv `model.confidence_type=quantile` oid. 
In plaats van hier een lijstje van modeltypes te definiëren.

Reply:
I disgree here, I think the prediction job should contain the configuration on what we want to happen for a specific prediction/ model. If we also add an attribute to the model we are moving away from the centralized config in the prediction job. I agree that keeping a list here on which confidence interval should be made for which model_type here is not elegant. Maybe it is better to keep the list somewhere else and generate a "confidence_interval_type" entry in the prediction job based on that list.

Reply:
@FrankKr does the model_type (which we call "model") not imply the confidence_type as well, e.g. a `xgb_quantile` model always has a confidence_type of quantile. If so, I think its better to just keep using the model_type to do the selection.

@JanMaartenvanDoorn if `pj["model"]` already an enum? If not you should either convert it or compare to `MLModelType.XGB_QUANTILE.value`

Reply:
@JanMaartenvanDoorn I don't think I completely understand your comment. The configuration of the type of confidence interval is not defined in the prediction job, but based on the modeltype, right?

Suppose in the future we would include for example and LGB_QUANTILE model. 
I think it would be elegant if then only in one place in the project code has to be added; in the code implementing that model type. Currently, it looks like also in other places this new model type has to be added to an if-check.
Perhaps we could also find somewhere in between that we check if the ModelType ends in _QUANTILE and agree that the name of quantile-type models should always end in quantile?

Reply:
@FrankKr if we have multiple variations of quantile models (in the future) which can use the same implementation to calculate the quantiles it would indeed be nice to make the selection on a shared attribute. There are different strategies to accomplish such a thing. I would not be too bothered to add a line or two for a new model_type (I will not happen that often I guess). But we could also use an  additional property on the prediction_job, .e.g model_class | model_group | model_type_class | model_type_group -> "quantile" or something similar. To treat a group of model_types in a certain way.

Reply:
Looking at the `ConfidenceIntervalApplicator` again. The `model` (RegressorMixin) is already an attribute. We could also make the selection based on a base class perhaps. But then we have to wrap a common base around the sklearn implementations I guess..